update/org.eclipse.update.configurator/src/org/eclipse/update/internal/configurator/Utils.java

```diff
@@ -200,44 +198,21 @@ public static String getNL() {
 		return getContext().getProperty(PROP_NL);
 	}
 
-	/**
-	 * Returns a number that changes whenever the set of installed plug-ins
-	 * changes. This can be used for invalidating caches that are based on
-	 * the set of currently installed plug-ins. (e.g. extensions)
-	 *
-	 * @see PlatformAdmin#getState()
-	 * @see org.eclipse.osgi.service.resolver.State#getTimeStamp()
-	 */
-	public static long getStateStamp() {
-		ServiceReference<PlatformAdmin> platformAdminReference = getContext().getServiceReference(PlatformAdmin.class);
-		if (platformAdminReference == null) {
-			return -1;
-		}
-		PlatformAdmin admin = getContext().getService(platformAdminReference);
-		return admin == null ? -1 : admin.getState(false).getTimeStamp();
-	}
-
 	/**
 	 * Return the resolved bundle with the specified symbolic name.
 	 *
 	 * @see PackageAdmin#getBundles(String, String)
 	 */
 	public static synchronized Bundle getBundle(String symbolicName) {
-		if (bundleTracker == null) {
-			bundleTracker = new ServiceTracker<>(getContext(), PackageAdmin.class, null);
-			bundleTracker.open();
-		}
-		PackageAdmin admin = bundleTracker.getService();
-		if (admin == null) {
-			return null;
-		}
-		Bundle[] bundles = admin.getBundles(symbolicName, null);
-		if (bundles == null) {
+		BundleContext context = getContext();
+		if (context == null) {
 			return null;
 		}
+		Bundle[] bundles = context.getBundles();
 		//Return the first bundle that is not installed or uninstalled
 		for (Bundle bundle : bundles) {
-			if ((bundle.getState() & (Bundle.INSTALLED | Bundle.UNINSTALLED)) == 0) {
+			if (bundle.getSymbolicName().equals(symbolicName)
+					&& (bundle.getState() & (Bundle.INSTALLED | Bundle.UNINSTALLED)) == 0) {
 				return bundle;
 			}
 		}
```

Comment:
Can't this entire method be just replaced with `org.eclipse.core.runtime.Platform.getBundle(String)`?

Reply:
I replaced all calls with a call to platform now.